### DISCUSSION TITLE
Third-party component adds <style> to <head>, which gets removed on re-render

### DISCUSSION THREAD

**Discussion by WickyNilliams** [vercel/next.js#16155] [Help] [answered]
Hey,

I have a third party component I am using in my Next.js app. This component adds a `<style>` tag to `<head>` (not via `next/head`, but via some mechanism internal to the component). The reasons for this are outside of my control.

When Next.js re-renders my `<head>` on say `<title>` change, the injected `<style>` tags are lost during reconciliation (presumably). These `<style>` tags are not known ahead of time, so i cannot server-render them to ensure they remain across re-renders.

Is there some way I can manage this, aside from never updating `<head>` at all?

**Comment by jamesmosier** [comment]
Hey @WickyNilliams. Do you have an example repository you could share?

**Comment by WickyNilliams** [comment]
@jamesmosier I tried creating a really minimal example, but wasn't able to reproduce there. I'll have to take another look at it when i get a chance. I'll close this for now

**Comment by hasaneks** [comment]
Hey @jamesmosier,  I need your help. I encounter was same problem but I don't find solution for this problem.

I use story widget script code. Web page : https://storywidget.com/tr/

**Comment by WickyNilliams** [comment]
Possibly relevant issue https://github.com/vercel/next.js/issues/11012

**Comment by zGrav** [comment, marked as answer]
This was happening to me today on the `index.ts` but not on a `[id].ts` page.

I tried to compare what was different between both, and I narrowed it down to having this: `<link rel="icon" href="/favicon.ico" />` on `index.ts` would break it.

Moving the whole `Head` to `_app.ts` no longer broke it.

**Reply by jorgefelico** [reply]
@WickyNilliams  I think this should be marked as the answer. This fixed the issue for me! Thank you so much @zGrav

**Reply by zGrav** [reply]
Glad to know, I wonder if I should potentially approach someone from Vercel to look into this to be honest.

**Reply by WickyNilliams** [reply]
Done. Not able to verify myself, but I'll take your word on it :)
